products/compute/api.yaml

```diff
@@ -11316,6 +11316,61 @@ objects:
           PARTNER InterconnectAttachment is created, updated,
           or deleted.
         output: true
+      - !ruby/object:Api::Type::Array
+        name: bfd
+        description: |
+          BFD configuration for the BGP peering.
+        min_version: beta
+        item_type: !ruby/object:Api::Type::NestedObject
+            properties:
+              - !ruby/object:Api::Type::Integer
+                name: minReceiveInterval
+                description: |
+                  The minimum interval, in milliseconds, between
+                  BFD control packets received from the peer router. The actual value
+                  is negotiated between the two routers and is equal to the greater of
+                  this value and the transmit interval of the other router. If set,
+                  this value must be between 100 and 30000. The default is 300.
+                default_value: 300
+              - !ruby/object:Api::Type::Integer
+                name: minTransmitInterval
+                description: |
+                  The minimum interval, in milliseconds, between
+                  BFD control packets transmitted to the peer router. The actual value
+                  is negotiated between the two routers and is equal to the greater of
+                  this value and the corresponding receive interval of the other
+                  router. If set, this value must be between 100 and 30000. The default
+                  is 300.
+                default_value: 300
+              - !ruby/object:Api::Type::Integer
+                name: multiplier
+                description: |
+                  The number of consecutive BFD packets that must be missed
+                  before BFD declares that a peer is unavailable. If set, the value
+                  must be a value between 2 and 16. The default is 3.
+                default_value: 3
+              - !ruby/object:Api::Type::Enum
+                name: sessionInitializationMode
+                description: |
+                  The BFD session initialization mode for
+                  this BGP peer. If set to `ACTIVE`, the Cloud Router will initiate the
+                  BFD session for this BGP peer. If set to `PASSIVE`, the Cloud Router
+                  will wait for the peer router to initiate the BFD session for this
+                  BGP peer. If set to `DISABLED`, BFD is disabled for this BGP peer. The
+                  default is `PASSIVE`.
+                values:
+                  - :ACTIVE
+                  - :DISABLED
+                  - :PASSIVE
+      - !ruby/object:Api::Type::Boolean
+        name: 'enable'
+        description: |
+          The status of the BGP peer connection. If set to FALSE,
+          any active session with the peer is terminated and all associated
+          routing information is removed. If set to TRUE, the peer connection
+          can be established with routing information. The default is TRUE.
```

Comment:
Let's put an explicit default here, that way the send_empty_value doesn't kick in on create to send a 'false' to the API.

Reply:
explicit default set to "true", but the API returns "TRUE" if not provided which cannot be unmarshalled into a Boolean (AFAIU, causing another panic), so also changed the attribute to String (maybe we should make it an Enum instead?)

Reply:
Oh no, that's not a great UX. It sounds like they want to retain the ability to have other enum values in the future, but I didn't get an impression that they have explicit plans around that, just that they've been burned by booleans in the past and want to be cautious. Our choices are:
1. Make it an enum and require users to say "FALSE" and "TRUE" (not great UX, but more flexible)
2. Make it a boolean and use a custom expander/flattener that calls some of the strconv utils to convert the value

What do you think the likelihood of a new enum value appearing there is? From an API perspective I could imagine them adding an "ONLY_IF_X" sort of value, but from a feature perspective I don't know enough about the product to make an informed judgment.

Reply:
I like the idea of 2 better. I'm consulting internally and then we decide.

Reply:
Went for 2.